standalone/src/main/java/io/delta/standalone/OptimisticTransaction.java

```diff
@@ -34,6 +34,12 @@
      * Modifies the state of the log by adding a new commit that is based on a read at the table's
      * latest version as of this transaction's instantiation. In the case of a conflict with a
      * concurrent writer this method will throw an exception.
+     * <p>
+     * Note: any {@link io.delta.standalone.actions.AddFile} with an absolute path within the table
+     * path will be updated to have a relative path (based off of the table path). Because of this,
+     * be sure to generate all {@link io.delta.standalone.actions.RemoveFile}s from
+     * {@link io.delta.standalone.actions.AddFile}s read from the Delta Log, and not the
+     * {@link io.delta.standalone.actions.AddFile}s created pre-commit.
```

Comment:
```suggestion
     * {@link io.delta.standalone.actions.RemoveFile}s created pre-commit.
```
Since you need to change this, could you update RemoveFile constructor like this:

```
    /**
     * Users should <b>not</b> construct {@link RemoveFile}s themselves, and should instead use one
     * of the various {@link AddFile#remove()} methods to instantiate the correct {@link RemoveFile}
     * for a given {@link AddFile} instance.
     *
     * @deprecated {@link RemoveFile} should be created from {@link AddFile#remove()} instead.
     */
    @Deprecated
    public RemoveFile(
```
This would show up in IDE at least.